agent/config/config.go

```diff
@@ -184,6 +184,7 @@ type Config struct {
 	CheckUpdateInterval              *string                  `json:"check_update_interval,omitempty" hcl:"check_update_interval" mapstructure:"check_update_interval"`
 	Checks                           []CheckDefinition        `json:"checks,omitempty" hcl:"checks" mapstructure:"checks"`
 	ClientAddr                       *string                  `json:"client_addr,omitempty" hcl:"client_addr" mapstructure:"client_addr"`
+	Config                           ConfigEntries            `json:"config,omitempty" hcl:"config" mapstructure:"config"`
```

Comment:
I suggest we name this a little more clearly - a key called `config` at the top of a generic config file is a little odd. But even more importantly I think it's important we make it obvious that these are not the source of truth but only bootstrap values for initial startup.

The downside is that, later we might consider an option in the config file (maybe per type or per entry) that explicitly indicates the intention is to treat the config file as the source of truth. If set, we'd just overwrite the current config entry every time config is reloaded rather than only do an "insert if not exists".

I propose the following syntax for now:

```hcl
config_entries {
  bootstrap {
    # hcl block syntax seems nice here as well as using same kind string as in CLI etc.
    # but I've not thought through what it means for parsing and JSON compat...
    proxy-defaults "global" {
       # ...
    }
  }
}
```

Later we could support allowing the same structure outside of the `bootstrap` block which would indicate source-of-truth-ness (and maybe cause an error for write via API too?).

Reply:
I implemented the changes. As for the HCL block syntax it maps to json like:

```json
"proxy_defaults": {
   "global": {
      ...
   }
}
```

You can actually specify it in the HCL like that too if you want

This also appears to be valid HCL that maps to the correct config:

```hcl
config_entries {
   bootstrap proxy_defaults global {
      foo = "bar"
      bar = 1.0
   }
}
```